src/main/docs/guide/producer/producerMethods/producerParameters/producerHeaders.adoc

```diff
@@ -1,7 +1,10 @@
 Headers can be set on the message with the link:{apimicronaut}messaging/annotation/Header.html[@Header] annotation applied to the method or an argument of the method. Apply the annotation to the method itself if the value is static for every execution. Apply the annotation to an argument of the method if the value should be set per execution.
 
+It is also possible to pass 'RabbitHeaders' object as method argument to pass a Map of Headers dynamically.
+
 snippet::io.micronaut.rabbitmq.docs.headers.ProductClient[tags="imports,clazz", project-base="docs-examples/example]
 
 <1> Headers can be defined at the class level and will apply to all methods. If a header is defined on the method with the same name as one on the class, the value on the method will be used.
 <2> Multiple annotations can be used to set multiple headers on the method or class level.
-<3> Headers can be set per execution. The name is inferred from the argument if the annotation value is not set. The value passed to the method will always be used, even if null.
\ No newline at end of file
+<3> Headers can be set per execution. The name is inferred from the argument if the annotation value is not set. The value passed to the method will always be used, even if null.
+<4> A key/value map of headers can be set with the 'RabbitHeaders' object.
```

Comment:
You can mention `MessageHeaders` here as well.

Reply:
Have changed to specify any implementation of MessageHeaders Interface.